runtime/compiler/control/CompilationThread.cpp

```diff
@@ -9273,7 +9273,7 @@ TR::CompilationInfo::compilationEnd(J9VMThread * vmThread, TR::IlGeneratorMethod
 
                         if (TR::Options::isAnyVerboseOptionSet(TR_VerboseCompileEnd, TR_VerbosePerformance, TR_VerboseCompFailure))
                            {
-                           TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE,"Failure while relocating for %s", comp->signature());
+                           TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE,"Failure while relocating for %s, return code = %d [%s]\n", comp->signature(), returnCode, compilationErrorNames[returnCode]);
```

Comment:
This will crash when you have an AOT load that failed to relocate a Debug Counter. That's my fault though; `TR_RelocationRecordDebugCounter::applyRelocation` returns -1, when it should've returned `compilationFailure`. Do you mind changing it so that it doesn't return -1 anymore?

Reply:
Good point; but IMO changing the return code of `TR_RelocationRecordDebugCounter::applyRelocation` is out of the scope that this PR tries to achieve. I've made update so that it won't crash even if the return code isn't any of the defined one.